app/models/host_facets/reported_data_facet.rb

```diff
@@ -1,4 +1,24 @@
 module HostFacets
   class ReportedDataFacet < Base
+    def self.populate_fields_from_facts(host, parser, type, source_proxy)
+      facet = host.reported_data || host.build_reported_data
+      facet.attributes = {
+        boot_time: parser.boot_timestamp,
+        virtual: parser.virtual,
+        ram: parser.ram,
+        sockets: parser.sockets,
+        cores: parser.cores,
+      }.compact
+      facet.save if facet.changed?
+    end
+
+    def boot_time=(val)
+      val = Time.at(val) if val.is_a?(Numeric)
```

Comment:
should we have a test that makes sure this is properly set when passing a numeric timestamp?

Reply:
I've moved the tests to separate file. Another thing that facets brings us :)

Reply:
And added the test for assignmet by seconds